source/slang/slang-ir-insts.h

```diff
@@ -3689,72 +3037,175 @@ struct IRBuilder
     IRInst* getCapabilityValue(CapabilitySet const& caps);
 
     IRBasicType* getBasicType(BaseType baseType);
-    IRBasicType* getVoidType();
-    IRBasicType* getBoolType();
-    IRBasicType* getIntType();
-    IRBasicType* getInt64Type();
-    IRBasicType* getUIntType();
-    IRBasicType* getUInt64Type();
-    IRBasicType* getUInt16Type();
-    IRBasicType* getUInt8Type();
-    IRBasicType* getFloatType();
-    IRBasicType* getCharType();
-    IRStringType* getStringType();
-    IRNativeStringType* getNativeStringType();
-    IRNativePtrType* getNativePtrType(IRType* valueType);
-
-    IRType* getCapabilitySetType();
-
-    IRAssociatedType* getAssociatedType(ArrayView<IRInterfaceType*> constraintTypes);
-    IRThisType* getThisType(IRType* interfaceType);
-    IRRawPointerType* getRawPointerType();
-    IRRTTIPointerType* getRTTIPointerType(IRInst* rttiPtr);
-    IRRTTIType* getRTTIType();
-    IRRTTIHandleType* getRTTIHandleType();
-    IRAnyValueType* getAnyValueType(IRIntegerValue size);
-    IRAnyValueType* getAnyValueType(IRInst* size);
-    IRDynamicType* getDynamicType();
 
-    IRTargetTupleType* getTargetTupleType(UInt count, IRType* const* types);
+#if 0 // FIDDLE TEMPLATE:
+%local ir_lua = require("source/slang/slang-ir.h.lua")
+%local basic_types = ir_lua.getBasicTypesForBuilderMethods()
+%for _, type_info in ipairs(basic_types) do
+%  -- Declare variables for both variadic and non-variadic cases
+%  local non_variadic_operands = {}
+%  local variadic_operand = nil
+%  
+%  -- For variadic types, separate operands by type
+%  if type_info.is_variadic then
+%    for _, op in ipairs(type_info.operands) do
+%      if op.variadic then
+%        variadic_operand = op
+%      else
+%        table.insert(non_variadic_operands, op)
+%      end
+%    end
+%  end
+%
+%  -- Unified approach: N lists with specified counts (1 for single operands, paramCount for variadic)
+%  if type_info.is_variadic then
+$(type_info.return_type) $(type_info.method_name)(
+%    for i, operand in ipairs(non_variadic_operands) do
+    $(operand.type)* $(operand.name),
+%    end
+    UInt $(variadic_operand.name)Count, $(variadic_operand.type)* const* $(variadic_operand.name))
+%  else
+$(type_info.return_type) $(type_info.method_name)(
+%    for i, operand in ipairs(type_info.operands) do
+    $(operand.type)* $(operand.name)$(operand.optional and " = nullptr" or "")
+%      if i < #type_info.operands then
+,
+%      end
+%    end
+)
+%  end
+{
+%  if #type_info.operands == 0 then
```

Comment:
I realize it would involve a frustrating amount of duplication, but it would be nice to move the implementations of these factory functions off to `slang-ir-insts.cpp`, so that we can reduce the amount of generator logic here in the `.h`.

Reply:
I wonder if we can implement some sort of parameterized template, so we can define it here and run it elsewhere